arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt

```diff
@@ -1,48 +1,90 @@
 package no.nav.soknad.arkivering.soknadsarkiverer.utils
 
-import no.nav.soknad.arkivering.soknadsarkiverer.config.*
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties
+import no.nav.soknad.arkivering.soknadsarkiverer.kafka.KafkaConfig
+import org.apache.commons.compress.harmony.pack200.PackingUtils.config
+import org.junit.jupiter.api.AfterAll
 import org.junit.jupiter.api.fail
-import org.springframework.context.annotation.Bean
-import org.springframework.context.annotation.Configuration
-import org.springframework.context.annotation.Primary
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.context.properties.bind.Binder
+import org.springframework.boot.context.properties.source.ConfigurationPropertySource
+import org.springframework.boot.context.properties.source.MapConfigurationPropertySource
+import org.springframework.core.io.ClassPathResource
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
 import org.testcontainers.containers.KafkaContainer
 import org.testcontainers.utility.DockerImageName
-import java.io.Closeable
+import java.util.Properties
 
-@Configuration
-class ContainerizedKafka : Closeable {
-	private val kafkaContainer: KafkaContainer = KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
-		.withNetworkAliases("kafka-broker")
 
-	init {
-		kafkaContainer.start()
-		listOf(kafkaMainTopic, kafkaProcessingTopic, kafkaMessageTopic, kafkaMetricsTopic)
-			.forEach { createTopic(it) }
-	}
+open class ContainerizedKafka  {
+
+
+
+	companion object {
+		private val kafkaConfig : KafkaConfig
+
+		init {
+			val factoryBean = YamlPropertiesFactoryBean()
+			factoryBean.setResources(ClassPathResource("application.yml"))
+
+			val properties: Properties? = factoryBean.getObject()
+
+			val propertySource: ConfigurationPropertySource = MapConfigurationPropertySource(properties)
```

Comment:
A bit of a matter of taste, but I'd remove the types. Especially when the type of `properties`  is `Properties?`, it doesn't add so much to explicitly state it, and the downside is extra cognitive load when each line is more packed with extra information.

Reply:
It was automatic convert from Java :D